core/src/main/java/org/apache/commons/vfs2/provider/AbstractFileSystem.java

```diff
@@ -317,42 +317,46 @@ public FileObject resolveFile(final FileName name) throws FileSystemException
         return resolveFile(name, true);
     }
 
-    private synchronized FileObject resolveFile(final FileName name, final boolean useCache) throws FileSystemException
```

Comment:
note that only the last conditional and contained method call is outside the synchronized block.  Alternatively the conditional could be done inside, setting a local variable, and just the refresh() call would be outside the block (though I don't imagine the cache strategy is likely to change mid-flight).  The point being the code is still thread-safe the former still within the synchronized block and the latter is synchronized on the underlying filesystem.

I don't have a reproducible test across multiple threads accessing different files on the same FS that reliably shows the issue.